prow/cmd/crier/main.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+
+	prowjobclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	prowjobinformer "k8s.io/test-infra/prow/client/informers/externalversions"
+
+	"k8s.io/test-infra/prow/crier"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+const (
+	resync = 30 * time.Second
+)
+
+// TODO(krzyzacy): copy & paste, refactor this
+// loadClusterConfig loads connection configuration
+// for the cluster we're deploying to. We prefer to
+// use in-cluster configuration if possible, but will
+// fall back to using default rules otherwise.
+func loadClusterConfig() (*rest.Config, error) {
+	clusterConfig, err := rest.InClusterConfig()
+	if err == nil {
+		return clusterConfig, nil
+	}
+
+	credentials, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+	if err != nil {
+		return nil, fmt.Errorf("could not load credentials from config: %v", err)
+	}
+
+	clusterConfig, err = clientcmd.NewDefaultClientConfig(*credentials, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load client configuration: %v", err)
+	}
+	return clusterConfig, nil
+}
+
+// TODO(krzyzacy): move this to kube???
+// getKubernetesClient retrieves the Kubernetes cluster
+// client from within the cluster
+func getKubernetesClient() (kubernetes.Interface, prowjobclientset.Interface) {
```

Comment:
You'll also need to ensure you are looking for ProwJobs in the right namespace: https://github.com/kubernetes/test-infra/blob/fa113fe0feed055303ca00b4e5267a1da33e2071/prow/config/config.go#L78-L82

Reply:
the pjClient can handle prowjob in all namespaces, and the lister can get prowjobs by taking in the namespace

```
c.informer.Lister().ProwJobs(namespace).Get(name)
```